### PR TITLE
xtest 20xxx: fix against deprecated CFG_ENC_FS

### PR DESCRIPTION
**Pull request by etienne-lms**
fix optee_test for changes in secure storage by [optee_os PR1078](https://github.com/OP-TEE/optee_os/pull/1078)

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Shouldn't we still test for `CFG_REE_FS` as the tests are only executed for that file system.

**Comment by etienne-lms** [comment]
xtest tests 60xx use CFG_REE_FS, CFG_RPMB_FS and CFG_SQL_FS to specifically test these.
But if none are defined, xtest 60xx still run secure storage test using the "default" secure storage support.

I fell test 20xxx should behave the same. The expect that at least a secure storage is supported and is to be tested.

An alternate is to enable xtest 60xx and 20xxx upon `CFG_REE_FE|CFG_RPMB_FS|CFG_SQL_FS`.

**Comment by jenswi-linaro** [comment]
Is this tested without `CFG_REE_FS` defined in optee_os?

**Comment by etienne-lms** [comment]
ok, my mistake: xtest 20xxx are specifically design for "REE FS" secure storage.

File: host/xtest/xtest_main.c [hunk 1]
```diff
@@ -111,7 +112,7 @@ ADBG_SUITE_ENTRY(XTEST_TEE_20503, NULL)
 ADBG_SUITE_ENTRY(XTEST_TEE_20521, NULL)
 ADBG_SUITE_ENTRY(XTEST_TEE_20522, NULL)
 ADBG_SUITE_ENTRY(XTEST_TEE_20523, NULL)
-#endif /* defined(CFG_ENC_FS) && defined(CFG_REE_FS) */
+#endif
```
Comment (jenswi-linaro) [inline host/xtest/xtest_main.c]: Please leave a comment indicating which `#if` this is ending

File: host/xtest/xtest_test.h [hunk 2]
```diff
@@ -99,7 +99,7 @@ ADBG_CASE_DECLARE(XTEST_TEE_20503);
 ADBG_CASE_DECLARE(XTEST_TEE_20521);
 ADBG_CASE_DECLARE(XTEST_TEE_20522);
 ADBG_CASE_DECLARE(XTEST_TEE_20523);
-#endif /* defined(CFG_ENC_FS) && !defined(CFG_ENC_FS) */
+#endif
```
Comment (jenswi-linaro) [inline host/xtest/xtest_test.h]: Please leave a comment indicating which `#if` this is ending

File: host/xtest/xtest_20000.c [hunk 3]
```diff
@@ -129,7 +129,7 @@ static TEEC_Result obj_open(TEEC_Session *sess, void *id, uint32_t id_size,
 	op.params[0].tmpref.size = id_size;
 	op.params[1].value.a = flags;
 	op.params[1].value.b = 0;
-	op.params[2].value.a = TEE_STORAGE_PRIVATE;
+	op.params[2].value.a = TEE_STORAGE_PRIVATE_REE;
```
Comment (jforissier) [inline host/xtest/xtest_20000.c]: This won't compile because `<tee_api_defines_extensions.h>` is not included.

Comment (etienne-lms) [inline host/xtest/xtest_20000.c]: great! 
I wonder how i validated this change !
My apologies.
However, i think this change is mandatory as the tests really needs REE_FS storage from optee.

Comment (jforissier) [inline host/xtest/xtest_20000.c]: Yes, I agree.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
(well My R-b assumes you add the missing `#include` of course ;-))

**Comment by etienne-lms** [comment]
i though your tag would prevent me from fixing my bugs!

**Comment by etienne-lms** [comment]
oups, i forgot to rebase...
here it comes.

**Pull request merged**